File: Problemset/linked-list-cycle/linked-list-cycle.java

// @Title: 环形链表 (Linked List Cycle)
// @Author: 15218859676
// @Date: 2020-08-19 11:53:07
// @Runtime: 0 ms
// @Memory: 40 MB

/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if(head == null || head.next == null){
            return false;
        }
        ListNode fast = head;
        ListNode slow = head;
        while(fast != null && fast.next != null){
            fast = fast.next.next;
            slow = slow.next;
            if(fast == slow ){
                return true;
            }
        }
        return false;
    }
}

File: Problemset/linked-list-cycle/linked-list-cycle.py

# @Title: 环形链表 (Linked List Cycle)
# @Author: 15218859676
# @Date: 2020-08-19 11:55:44
# @Runtime: 40 ms
# @Memory: 19 MB

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        fast = head
        slow = head
        while fast  and  fast.next :
            fast = fast.next.next
            slow = slow.next
            if fast == slow:
                return True
        return False
